Fix log*_throttle with sim time

salihmarangoz opened the pull request:
When log*_throttle is used with bags or a simulation, logging stops when bag/simulation restarts. Solved this problem with resetting logging_time_table when ros time moved backward.

Testing code:
```py
import rospy

rospy.init_node('test_node')

while not rospy.is_shutdown():
    rospy.loginfo_throttle(100.0, "Throttled message!")
```

salihmarangoz committed:
Fix log*_throttle with sim time

When log*_throttle is used with bags or a simulation, logging stops. Solved this problem with resetting logging_time_table when ros time moved backward

dirk-thomas commented on clients/rospy/src/rospy/core.py:
```diff
@@ -223,6 +223,9 @@ def __call__(self, caller_id, period):
               (now - last_logging_time) > rospy.Duration(period)):
             self.last_logging_time_table[caller_id] = now
             return True
+        elif (last_logging_time > now):
```
Nitpick: please remove the parenthesis.

salihmarangoz replied:
Done, thanks

dirk-thomas commented on clients/rospy/src/rospy/core.py:
```diff
@@ -223,6 +223,9 @@ def __call__(self, caller_id, period):
             self.last_logging_time_table[caller_id] = now
             return True
+        elif (last_logging_time > now):
+            self.last_logging_time_table = {}
```
After resetting the dictionary shouldn't the `caller_id` still be added: `self.last_logging_time_table[caller_id] = now`?

salihmarangoz replied:
Yes, you are right. I fixed now it should prevent losing a log message when the time goes back

dirk-thomas replied:
I don't think it would have lost a message but always also printed the next message. Anyway its fixes. Thanks.

salihmarangoz committed:
fixed a bug where client loses a log message when rostime goes backward

removed paranthesis
fixed a bug where client loses a log message when rostime goes backward

salihmarangoz commented:
It should be ok now

dirk-thomas commented:
Thanks for the patch.
